Sources/MapboxCoreNavigation/Router.swift

```diff
@@ -290,7 +290,8 @@ extension InternalRouter where Self: Router {
         }
         isRefreshing = true
         resolvedRoutingProvider.refreshRoute(indexedRouteResponse: indexedRouteResponse,
-                                             fromLegAtIndex: UInt32(legIndex)) { [weak self] session, result in
+                                             fromLegAtIndex: UInt32(legIndex),
+                                             currentRouteShapeIndex: shapeIndex) { [weak self] session, result in
```

Comment:
As you plumb this through further, one tricky consideration is that, last I checked, the API automatically limits the number of segments returned when passing in a shape index. In other words, the API will only refresh part of a long route. We’ll need to clear out any subsequent attributes.